src/core/thread/mle.cpp

```diff
@@ -3403,6 +3483,8 @@ void Mle::HandleChildUpdateRequest(RxInfo &aRxInfo)
         ExitNow(error = kErrorParse);
     }
 
+    aRxInfo.mClass = (challenge.mLength == 0) ? RxInfo::kPeerMessage : RxInfo::kTentativeMessage;
```

Comment:
Should a Child Update Request from a valid child be considered a peer message regardless of whether the Challenge TLV is included?

Reply:
I proposed in SPEC-1025 to have the Child Update Request always be treated as a peer MLE message. That would simplify this line to:

    `aRxInfo.mClass = RxInfo::kPeerMessage;`

The behavior would be fully equivalent I think, because if the peer message is processed and the originator ("neighbor") has IsStateValid() == false, then the particular tKSC advancement code for peer messages would not be executed anyway. So then the processing becomes equivalent to tentative class. So it seems not needed to change the class dynamically to "tentative" here in some cases, as the class of "peer" already includes the assumption that if you receive such message from a non-peer i.e. unknown node or invalid node then don't treat it; and for sure don't update your tKSC based on it.

Reply:
Thanks @EskoDijk . Yes. I agree. Treating it as peer message works and would be simpler as well (since peer class already has the check for being from a valid neighbor). I changed it in the PR

Reply:
What about other messages like Link Request? Should those always be treated as peer as well?

Reply:
interesting question. 

Can we even consider removing "tentative" class fully and use "peer" for the cases where we now use "tentative".
W use "tentative" for ParentRequest, LinkRequest or Announce (from list below). In all cases, I think treating it as peer would may work fine (peer classification will not adopt a higher key seq if the neighbor/sender is not a valid neighbor which what tentative would do). This can potentially help simplify the classification (have only two classes instead of 3). Thoughts?  @jwhui @EskoDijk 

```
Advertisement                  Peer
DataRequest                    Peer
DataResponse                   Peer
ParentRequest                  Tentative
ParentResponse                 Authoritative      (always includes the ResponseTlv)
ChildIdRequest                 Authoritative      (always includes the ResponseTlv)
ChildIdResponse                Peer
Announce                       Tentative
ChildUpdateRequest             Peer 
ChildUpdateResponse            Authoritative if has ResponseTlv, Peer otherwise.
LinkRequest                    Tentative
LinkAccept                     Authoritative     (only if has ResponseTlv)
LinkAcceptAndRequest           Authoritative     (only if has ResponseTlv)
LinkMetricsManagementRequest   Peer
LinkMetricsManagementResponse  Peer
LinkProbe                      Peer
TimeSync                       Peer              (OT only for Time Sync feature).
```

Reply:
Historically the 'tentative' class was needed:
* Peer messages are not processed at all ('SHOULD NOT') when delta-tKSC > 1. 
* Tentative messages are processed (called 'partially processed') when delta-tKSC > 1. 
* So the discovery type messages like Link Request / Parent Request / Announce would work even if the recipient was way behind with tKSC.  Useful to merge back long-separated partitions for example.

But now with new text we updated the 'peer' class to allow some processing when delta-tKSC > 1, as long as it doesn't change internal state. So that would allow all the discovery type messages to be processed and responses to be sent back.

This seems to remove one of the biggest reasons to have the 'tentative' class. Another minor reason might be: 
* current 'tentative' messages are all typically followed by further actions using MLE (unicast) messages; 
* effectively postponing any tKSC updates to later during the interaction.  
* While I do see the logic of this (don't let a discovery type message cause side-effects);
* it doesn't seem to have much practical benefits.